### PR TITLE
net-nds/phpldapadmin-1.2.6.6: version bump

### PR DESCRIPTION
**Pull request by protegh**
Closes: https://bugs.gentoo.org/897680

### CODE REVIEW

**Comment by protegh** [comment]
Closes: https://bugs.gentoo.org/897680

File: net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch [hunk 1]
```diff
@@ -0,0 +1,102 @@
+--- phpldapadmin/lib/PageRender.php_orig	2020-11-20 11:57:35.293875838 +0100
```
Comment (ceamac) [inline net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch:1]: Please add relevant comments, links etc at the top of each patch.  They should contain at least a short description of the problem they are trying to solve and a link to the issue or upstream patch.

File: net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
++				return $matchingtmplIDs[0];         # so we use it.
++			elseif (count($unmatchingtmplIDs) == 0) # No template found at all,
++				return 'none';                      # so we use the default template.
++			else                                    # Multiple templates found,
++				$this->drawTemplateChoice();        # so propose the template choice.
+ 		} else {
+ 			if (DEBUGTMP) printf('<font size=-2>%s:<u>%s</u></font><br />',__METHOD__,'SELECT a template to use.');
+ 
+--- phpldapadmin/lib/Template.php_orig	2020-11-20 11:57:35.276875606 +0100
++++ phpldapadmin/lib/Template.php	2021-11-18 15:50:05.381355470 +0100
```
Comment (ceamac) [inline net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch:37]: Usually we replace the file names with `a/lib/Template.php` and `b/lib/Template.php`, and also remove the extra info (timestamps are irrelevant).

You can use `scrub-patch` from `app-portage/iwdevtools`, it helps.

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+HOMEPAGE="https://github.com/leenooks/phpLDAPadmin"
+SRC_URI="https://github.com/leenooks/${MY_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+KEYWORDS="~alpha ~amd64 ~ia64 ~ppc ~sparc ~x86"
+IUSE=""
+
+RDEPEND="
+	>=dev-lang/php-8.0[hash(+),ldap,session,xml,nls]
+	dev-libs/openssl
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:19]:
```suggestion
	dev-libs/openssl:=
```
This way, this package will get reinstalled when switching from openssl:0/1.1 to openssl:0/3, which will help with the conditional patching.

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+inherit webapp
+
+MY_PN="phpLDAPadmin"
+DESCRIPTION="phpLDAPadmin is a web-based tool for managing all aspects of your LDAP server"
+HOMEPAGE="https://github.com/leenooks/phpLDAPadmin"
+SRC_URI="https://github.com/leenooks/${MY_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+KEYWORDS="~alpha ~amd64 ~ia64 ~ppc ~sparc ~x86"
+IUSE=""
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:15]: Please remove empty `IUSE`

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	"${FILESDIR}/${PN}-1.2.6.2-r1-default-templates.patch"
+	"${FILESDIR}/${PN}-1.2.6.4-getDN-htmlspecialchars.patch"
+)
+
+need_httpd_cgi
+
+src_prepare() {
+	has_version "dev-libs/openssl:0/1.1" && \
+		eapply "${FILESDIR}/${PN}-1.2.6.5-openssl-1.patch"
+	mv config/config.php.example config/config.php
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:35]:
```suggestion
	mv config/config.php.example config/config.php || die
```

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	default
+	# fix QA notice about broken IDAT window length
+	pngfix --out=network.png htdocs/images/default/network.png
+	pngfix --out=document.png htdocs/images/default/document.png
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:39]: You need to add `media-libs/libpng` to `BDEPEND` for this.

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+need_httpd_cgi
+
+src_prepare() {
+	has_version "dev-libs/openssl:0/1.1" && \
+		eapply "${FILESDIR}/${PN}-1.2.6.5-openssl-1.patch"
+	mv config/config.php.example config/config.php
+	default
+	# fix QA notice about broken IDAT window length
+	pngfix --out=network.png htdocs/images/default/network.png
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:38]:
```suggestion
	pngfix --out=network.png htdocs/images/default/network.png || die
```

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+	pngfix --out=document.png htdocs/images/default/document.png
+	mv -f network.png document.png htdocs/images/default/
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:40]:
```suggestion
	mv -f network.png document.png htdocs/images/default/ || die
```

**Review by ceamac — CHANGES_REQUESTED** [review]
I only did a quick review, I'll try to find more time to check this fully.

For the moment please also add the Manifest, [the CI is unhappy](https://qa-reports.gentoo.org/output/gentoo-ci/e556ec68f5/output.html#net-nds/phpldapadmin).

Thank you!

File: net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch (hunk 1 above)
Comment (protegh) [reply net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch]: I added some comments at the top of the patches.

File: net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch (hunk 2 above)
Comment (protegh) [reply net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch]: I changed filenames to the "a b" form without extra info.

**Comment by protegh** [comment]
I implemented all your other suggestions and uploaded the new files (including the Manifest now).

**Comment by protegh** [comment]
One question: Can you tell me why the gentoo bot didn't link the bug https://bugs.gentoo.org/897680 with this PR? When I created the PR I hadn't the "Closes: ..." line in the commit message, but then I added it and appended the "[please reassign] to the pull request title, but that didn't help.

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 9]
```diff
@@ -7,29 +7,39 @@ inherit webapp
...
+BDEPEND="
+	media-libs/libpng
+"
 S="${WORKDIR}/${MY_PN}-${PV}"
 
-# http://phpldapadmin.git.sourceforge.net/git/gitweb.cgi?p=phpldapadmin/phpldapadmin;a=commit;h=7dc8d57d6952fe681cb9e8818df7f103220457bd
 PATCHES=(
 	"${FILESDIR}/${PN}-1.2.1.1-fix-magic-quotes.patch"
+    "${FILESDIR}/${PN}-1.2.6.2-r1-default-templates.patch"
+    "${FILESDIR}/${PN}-1.2.6.4-getDN-htmlspecialchars.patch"
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:29]: These 2 lines should start with a tab, not 4 spaces.

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 10]
```diff
@@ -7,29 +7,39 @@ inherit webapp
...
+BDEPEND="
+	media-libs/libpng
+"
 S="${WORKDIR}/${MY_PN}-${PV}"
 
-# http://phpldapadmin.git.sourceforge.net/git/gitweb.cgi?p=phpldapadmin/phpldapadmin;a=commit;h=7dc8d57d6952fe681cb9e8818df7f103220457bd
 PATCHES=(
 	"${FILESDIR}/${PN}-1.2.1.1-fix-magic-quotes.patch"
+	"${FILESDIR}/${PN}-1.2.6.2-r1-default-templates.patch"
+	"${FILESDIR}/${PN}-1.2.6.4-getDN-htmlspecialchars.patch"
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:29]: This looks like a small enough bugfix, I will merge it but please sent it upstream too.

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild [hunk 11]
```diff
@@ -7,29 +7,39 @@ inherit webapp
...
 )
 
 need_httpd_cgi
 
 src_prepare() {
-	mv config/config.php.example config/config.php
+	has_version "dev-libs/openssl:0/1.1" && \
+		eapply "${FILESDIR}/${PN}-1.2.6.5-openssl-1.patch"
```
Comment (ceamac) [inline net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:36]: Not a fan of conditional patching, but since openssl-1.1 is on its way out I think we can keep this here for a few months.

But why is this needed?  By reading the linked issue, it seems they fixed it with aes-256-gcm and the fix is included in 1.2.6.6.  Can you test if 1.2.6.6 works with openssl 1.1 without the patch?

**Review by ceamac — CHANGES_REQUESTED** [review]
Thank you for your work on this.  There are a few issues left:
- please drop `1.2.6.2-r1-default-templates.patch` (maybe try to upstream it)
- please check if `1.2.6.5-openssl-1.patch` can also be dropped
- please try to squash the commits.  Ideally we should have 1 commit adding 1.2.6.6 and one commit dropping 1.2.6.3.  Then `git push --force` to replace them here.

Or I can do this if you find it too daunting, it's not a problem.  Just let me know.

**Comment by protegh** [comment]
> There are a few issues left:
> 
>     * please drop `1.2.6.2-r1-default-templates.patch` (maybe try to upstream it)
Upstream PR: https://github.com/leenooks/phpLDAPadmin/pull/219

>     * please check if `1.2.6.5-openssl-1.patch` can also be dropped
As https://github.com/leenooks/phpLDAPadmin/issues/183#issuecomment-1450946719 clearly states that the upstream fix works with openssl-1 we can drop this patch.

>     * please try to squash the commits.  Ideally we should have 1 commit adding 1.2.6.6 and one commit dropping 1.2.6.3.  Then `git push --force` to replace them here.
> 
> 
> Or I can do this if you find it too daunting, it's not a problem. Just let me know.
Please do so.

File: net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild (hunk 10 above)
Comment (protegh) [reply net-nds/phpldapadmin/phpldapadmin-1.2.6.6.ebuild:29]: Note: The file `lib/HTMLTree.php` that si changed by this patch was removed in this upstream commit: https://github.com/leenooks/phpLDAPadmin/commit/5b046a95eb3cfd696b0f9434b99792dab37022a0

So future versions won't need this patch anymore.

**Review by ceamac — APPROVED** [review]
I squashed the commits and tidied it up a little, I'll merge it.  Thank you!
